### PR TITLE
fixed tensorflow.math.softplus

### PR DESCRIPTION
**Pull request by ZJay07**
# PR Description
image

## Related Issue

Close https://github.com/unifyai/ivy/issues/28465

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

<!--
Please mark your PR as a draft if you realise after the fact that your tests are not passing or
that your pre-commit check has some failures.

Here are some relevant resources regarding tests and pre-commit:

https://unify.ai/docs/ivy/overview/deep_dive/ivy_tests.html
https://unify.ai/docs/ivy/overview/deep_dive/formatting.html#pre-commit

### CODE REVIEW

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi @ZJay07 
Thank you very much for the PR
I think there are other dtypes that the native tf does support like tf.float16
as maybe checked with
```python
import tensorflow as tf
x = tf.constant([0.5], dtype=tf.float16)
tf.math.softplus(x)
```
does the test not pass for these?

**Comment by ZJay07** [comment]
Hey @Ishticode, I tried rerunning the test again adding float16 and got this error, does this means tf.math.softplus does not support float16? 

```
ivy.utils.exceptions.IvyBackendException: paddle: softplus: paddle: abs: (NotFound) The kernel with key (CPU, Undefined(AnyLayout), float16) of kernel `abs` is not registered. Selected wrong DataType `float16`. Paddle support following DataTypes: int32, int64, float64, complex128, float32, complex64.
```

**Comment by Ishticode** [comment]
> Hey @Ishticode, I tried rerunning the test again adding float16 and got this error, does this means tf.math.softplus does not support float16?
> 
> ```
> ivy.utils.exceptions.IvyBackendException: paddle: softplus: paddle: abs: (NotFound) The kernel with key (CPU, Undefined(AnyLayout), float16) of kernel `abs` is not registered. Selected wrong DataType `float16`. Paddle support following DataTypes: int32, int64, float64, complex128, float32, complex64.
> ```

This mean the paddle backend is unable to run this function with float16. You will need to specify float16 as unsupported for ivy's softplus function paddle backend. Does that make sense?

**Review by Ishticode — CHANGES_REQUESTED** [review]
This error means the paddle backend is unable to run this function with float16. You will need to specify float16 as unsupported for ivy's softplus function paddle backend. Does that make sense?

**Comment by ZJay07** [comment]
Just fixed it and including float16, is this the right way to do it?

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 1]
```diff
@@ -2858,6 +2858,8 @@ def test_tensorflow_softplus(
     on_device,
 ):
     input_dtype, x = dtype_and_x
+    if backend_fw == "paddle" and "float16" in input_dtype:
```
Comment (Ishticode) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:2861]: can this be specified as unsupported in paddle backends softplus implementation?

**Review by Ishticode — APPROVED** [review]
Thank you. Looks good to merge :)

**Pull request merged**